src/dfa/CM_Global.cpp

```diff
@@ -355,51 +355,47 @@ int   get_fid (char *arg, char *dir, char *fn, char *ft)
 
 ///////////////////////////////////////////////////////////////////////////////
 
-char* alloc_debug (int n_bytes, const char *fname, size_t lineno)
-{
-   char *x = (char *)malloc(n_bytes);
-   if (x == NULL)
-   {
-      n_errors++;
-      prt_log ("%s:%s: Allocation failure; %u bytes not available.\n\n",
-               fname, lineno, n_bytes);
-      Quit ();
-   }
-   memory_usage += n_bytes;
-   if (memory_usage > memory_max) memory_max = memory_usage;
-   return (x);
-}
+#ifdef _DEBUG
```

Comment:
I'd prefer to not have conditional compilation.  I've seen too many cases where code rots because a branch of the conditional is never exercised.

This is the reason that there are two versions of the memory allocation routines in the code that I wrote.  The compiler & linker will not include the unused version in the executable.  If it is included, then the options are set wrong.

Paul preferred the debug version to have the file and line number printed in the debug version to help find the location where the memory failure occured.
Can that be preserved?

Reply:
That's fine to me.